include/deal.II/base/tensor_base.h

```diff
@@ -1,6 +1,7 @@
 #ifndef dealii__tensor_base_h
 #define dealii__tensor_base_h
 #warning This file is deprecated. Use <deal.II/base/tensor.h> instead.
-#endif
 
 #  include <deal.II/base/tensor.h>
+
+#endif
```

Comment:
as long as there is guard in `base/tensor.h`, this should not make a difference.